### PR TITLE
43 rework backwards geometry for flexible studies

### PR DESCRIPTION
**Pull request by simonge**
### Briefly, what does this PR introduce?
Complete reworking of the far backwards geometry with variable positioning of tagger stations possible inside and outside of the main beam vacuum. Constructed from boolean geometries to allow the extended beamline vacuum and box to fit together correctly. Tagger stations defined in separate xml to be constructed from additional technology options added at a later date.

Documentation/comments will need to be added to explain how to adjust the geometry.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #43 )
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
Changes the readout structure for the backwards taggers, this might need to be changed again.
The Lumi system has for now been commented out in the xml file until it has been reworked on.

### Does this PR change default behavior?

### CODE REVIEW

File: ip6/far_backward/definitions.xml [hunk 1]
```diff
@@ -74,6 +74,59 @@
     <constant name="Q3PR_EndX"            value="0.987412766073207715*m"/>
     <constant name="Q3PR_Theta"           value="CrossingAngle"/>
 
+    <!------------------------------------------->
+    <!-- Parameters for backwards vacuum box   -->
+    <!------------------------------------------->
```
Comment (wdconinc) [inline ip6/far_backward/definitions.xml:79]: From https://www.w3.org/TR/REC-xml/#sec-comments: 'the string " -- " (double-hyphen) must not occur within comments.' That is why the LCG job fails.

File: src/BackwardsTracker_geo.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    
+//     Box    Tag_Wall_Box(wall/2,tag_h, airThickness/2);
+//     Volume wallVol("TagWallVolume", Tag_Wall_Box, Steel);
+//     wallVol.setVisAttributes("WhiteVis");
+//     Tagger_Air.placeVolume(wallVol, Position(vac_w-wall/2, 0, 0));
+    taggerAssembly.placeVolume(Tagger_Air,Position(0,0,tagboxL-airThickness/2));
+    
+  }
+    std::cout << airThickness << " " << vacuumThickness << std::endl;
+
```
Comment (wdconinc) [inline src/BackwardsTracker_geo.cpp:85]: No debugging output without DD4hep `printout` calls.

File: src/BackwardsTracker_geo.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  //return mdet;
+
+  
+    
+  PlacedVolume pv_mod = mdet.volume().placeVolume(taggerAssembly);
+  //PlacedVolume pv_mod = desc.pickMotherVolume(det).placeVolume(Tagger_Air, Position(0, 0, -tagboxL/2 + airThickness/2));
+  for(auto child: mdet.children()){
+    cout << child.second.name() << endl;
+  }
+  cout << det.parent().name() << endl;
```
Comment (wdconinc) [inline src/BackwardsTracker_geo.cpp:144]: No debugging output without DD4hep `printout` calls.

Comment (simonge) [reply src/BackwardsTracker_geo.cpp]: Removed the temporary debugging prints.

**Comment by simonge** [comment]
Any idea what I've broken @wdconinc? the checkGeometry command appears to work correctly in both my environment and the cvmfs

**Comment by wdconinc** [comment]
"Replace all `getAttrOrDefault` *on strings* by explicit `getAttrOrDefault<string>` for the older DD4hep version in LCG." [https://eic.cloud.mattermost.com/main/pl/h5gfijb19jg6pcwb4zhy8dq8fe]

**Comment by simonge** [comment]
Passes all checks but seems EICrecon has a more stringent restrictions on the geometry than the simulation. The current implementation of the taggers needs to be adjusted again.

**Comment by wdconinc** [comment]
Is this functionally ready for merging? Or are there more changes you wish to include in it?

**Comment by simonge** [comment]
> Is this functionally ready for merging? Or are there more changes you wish to include in it?

Functionally it appears fine. The implementation of the vacuum geometry currently doesn't appear to get exported correctly using dd_web_display so I'm investigating that first.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**